### PR TITLE
Fix stories spurious 404 errors, allow stories to be configurable per-course

### PR DESCRIPTION
**Pull request by RichDom2185**
### Description

Relevant backend PR: https://github.com/source-academy/backend/pull/1053 (must be merged before this can be merged)

Please refer to commits for full list of changes.

Allows stories state (enabled/disabled) to be configured on a per-course basis. When it is disabled, the frontend won't attempt to contact the stories backend for that particular course, thus removing the cause of the 404 error.

Also moves stories' routes under `/courses/:courseId`, since it is now course-dependent.

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Code quality improvements

### How to test

Everything still works.

### Checklist

- [x] I have tested this code
- [ ] I have updated the documentation

### CODE REVIEW

**Review by sayomaki — APPROVED** [review]
LGTM, requested changes have been fixed as discussed in person. Thanks!

Currently waiting for backend PR to merge before we proceed to merge this.

**Comment by RichDom2185** [comment]
> LGTM, requested changes have been fixed as discussed in person. Thanks!
> 
> Currently waiting for backend PR to merge before we proceed to merge this.

Backend PR has been merged! I'll set this to merge on successful CI. Thanks!

**Pull request merged**
